### PR TITLE
Streaming api for movies

### PR DESCRIPTION
**Pull request by Capn05**
Added Streaming Api to bookmarked recommendations, this way if you want to watch one of the recs we picked, you can see where its available.

Added API key for access
in fetchtmdbid.js: script to string match Movie titles from database to movie TMDB movie Ids
This was necessary because all streaming availability APIS either have IMDB or TMDB id as input
In prisma files: added TMDB Id as field in user table
in moviecard.jsx: Added fetch and logic for displaying movie streaming availability
Screenshot 2024-07-23 at 10 15 49 PM

### CODE REVIEW

File: fetchTMDBID.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
+import fetch from "node-fetch";
+import { PrismaClient } from "@prisma/client";
+import Bottleneck from "bottleneck";
+import dotenv from "dotenv";
+dotenv.config();
+
+const prisma = new PrismaClient();
+const apiKey = process.env.VITE_API_KEY; // Ensure your API key is loaded from .env
+
+// Initialize the limiter to handle up to 50 requests per second
```
Comment (samyalihamad) [inline fetchTMDBID.js:10]: Nice adding limiter

File: fetchTMDBID.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+const limiter = new Bottleneck({
+  minTime: 100,
+  maxConcurrent: 1,
+  highWater: 10000,
+});
+
+async function fetchAndUpdateMovie(movie) {
+  const cleanedTitle = movie.title.slice(0, -6);
```
Comment (samyalihamad) [inline fetchTMDBID.js:18]: just curious why the title needs to be cleaned?

File: fetchTMDBID.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    method: "GET",
+    headers: {
+      accept: "application/json",
+      Authorization: `Bearer ${apiKey}`,
+    },
+  };
+
+  const response = await fetch(url, options);
+  const data = await response.json();
+  const movieData = data.results[0];
```
Comment (samyalihamad) [inline fetchTMDBID.js:32]: Can `data.results` be null?

File: fetchTMDBID.js [hunk 4]
```diff
@@ -0,0 +1,59 @@
+import fetch from "node-fetch";
```
Comment (samyalihamad) [inline fetchTMDBID.js:1]: Is this a one time import script?

File: .env
Comment (kevinzhu00) [inline .env:1]: Generally, it is bad practice to directly commit a .env file. Usually, these files are added to the gitignore and need to be manually generated/build by the user of the code. The reason being is that the API keys are usually sensitive and should not be shared.

File: fetchTMDBID.js (hunk 2 above)
Comment (Capn05) [reply fetchTMDBID.js:18]: So the titles in the database include a data year (e.g. (2004)) which messes up the search query when trying to find TMDB IDs

File: fetchTMDBID.js (hunk 3 above)
Comment (Capn05) [reply fetchTMDBID.js:32]: I think TMDB Api returns an empty array if no results found

File: fetchTMDBID.js (hunk 4 above)
Comment (Capn05) [reply fetchTMDBID.js:1]: Yes

**Pull request merged**